api/v1alpha1/etcdpeer_types.go

```diff
@@ -81,6 +143,88 @@ type EtcdPeerList struct {
 	Items           []EtcdPeer `json:"items"`
 }
 
+var _ webhook.Validator = &EtcdPeer{}
+
+// ValidateCreate validates that all required fields are present and valid.
+func (o *EtcdPeer) ValidateCreate() error {
+	path := field.NewPath("spec")
+	var allErrs field.ErrorList
+
+	allErrs = append(
+		allErrs,
+		o.Spec.Storage.validate(path.Child("storage"))...,
+	)
+	return allErrs.ToAggregate()
+}
+
+// ValidateCreate validates that deletion is allowed
+// TODO: Not yet implemented
+func (o *EtcdPeer) ValidateDelete() error {
+	var allErrs field.ErrorList
+	return allErrs.ToAggregate()
+}
+
+// ValidateCreate validates that only supported fields are changed
+// TODO: Not yet implemented
+func (o *EtcdPeer) ValidateUpdate(old runtime.Object) error {
+	var allErrs field.ErrorList
+	return allErrs.ToAggregate()
+}
+
+var _ webhook.Defaulter = &EtcdPeer{}
+
+// Default sets default values for optional EtcdPeer fields.
+// This is used in webhooks and in the Reconciler to ensure that nil pointers
+// have been replaced with concrete pointers.
+// This avoids nil pointer panics later on.
+func (o *EtcdPeer) Default() {
+	if o.Spec.Storage != nil {
+		o.Spec.Storage.setDefaults()
+	}
+}
+
+// ExampleEtcdPeer returns a valid example for testing purposes
+func ExampleEtcdPeer(namespace string) *EtcdPeer {
```

Comment:
This shouldn't be in the `apis` package - should be in some internal/testdata package. We don't want to expose this as part of our API/export it to other users.

Reply:
Done.